src/openttd.cpp

```diff
@@ -1443,23 +1447,35 @@ void StateGameLoop()
 	assert(IsLocalCompany());
 }
 
-static IntervalTimer<TimerGameCalendar> _autosave_interval({TimerGameCalendar::MONTH, TimerGameCalendar::Priority::AUTOSAVE}, [](auto)
+/** Interval for regular autosaves. Initialized at zero to disable till settings are loaded. */
+static IntervalTimer<TimerGameAutosave> _autosave_interval(std::chrono::milliseconds::zero(), [](auto)
 {
-	if (_settings_client.gui.autosave == 0) return;
-	if ((_cur_month % _autosave_months[_settings_client.gui.autosave]) != 0) return;
+	/* We reset the command-during-pause mode here, so we don't continue
+	 * to make auto-saves when nothing more is changing. */
+	_pause_mode &= ~PM_COMMAND_DURING_PAUSE;
 
 	_do_autosave = true;
 	SetWindowDirty(WC_STATUS_BAR, 0);
+
+	static FiosNumberedSaveName _autosave_ctr("autosave");
+	DoAutoOrNetsave(_autosave_ctr);
+
+	_do_autosave = false;
```

Comment:
Is this global still needed ?

Reply:
Yup! A grep shows two usages of this variable, which this PR will not address.